File: odroid/Control/src/bert/control/model/LinkPointType.java
/**
 * Copyright 2019. Charles Coughlin. All Rights Reserved.
 *                 MIT License.
 */
package bert.control.model;

/**
 * These are the possible types of "LinkPoints" or
 * connection locations on a link.
 */
public enum LinkPointType
{
	APPENDAGE,
	ORIGIN,
	REVOLUTE
	;


	/**
	 * @return  a comma-separated list of all block states in a single String.
	 */
	public static String names()
	{
		StringBuffer names = new StringBuffer();
		for (LinkPointType type : LinkPointType.values())
		{
			names.append(type.name()+", ");
		}
		return names.substring(0, names.length()-2);
	}
}

File: Bert/control/src/main/kotlin/chuckcoughlin/bert/control/model/LinkPointType.kt
/**
 * Copyright 2019. Charles Coughlin. All Rights Reserved.
 * MIT License.
 */
package chuckcoughlin.bert.control.model

/**
 * These are the possible types of "LinkPoints" or
 * connection locations on a link.
 */
enum class LinkPointType {
    APPENDAGE, ORIGIN, REVOLUTE;

    companion object {
        /**
         * @return  a comma-separated list of all block states in a single String.
         */
        fun names(): String {
            val names = StringBuffer()
            for (type in values()) {
                names.append(type.name + ", ")
            }
            return names.substring(0, names.length - 2)
        }
    }
}